File: benchmark/read_gbq.go
package main

import (
	"fmt"
	"gpandas"
	"time"

	"github.com/joho/godotenv"
)

func main() {
	envFile, _ := godotenv.Read(".env")
	table_id := envFile["TABLEID"]
	start := time.Now()
	gp := gpandas.GoPandas{}
	df, err := gp.From_gbq("SELECT * FROM "+table_id, "jm-ebg")
	if err != nil {
		fmt.Printf("Error reading from gbq: %v\n", err)
		return
	}
	fmt.Println(df)
	end := time.Since(start)
	fmt.Printf("%f\n", end.Seconds())
}

File: benchmark/read_gbq.py
import time
from pandas_gbq import read_gbq
import os
from dotenv import load_dotenv
import certifi

def main():
    load_dotenv() 
    table_id = os.getenv("TABLEID")
    start = time.time()
    
    query = "SELECT * FROM " + table_id
    df = read_gbq(query, "jm-ebg")
    end = time.time() - start
    print(df)
    
    print(f"{end:.6f}")

if __name__ == "__main__":
    main()
